Added missing namespace and includes

ChWick opened the pull request:
Hi, its me again!
When using my custom build files based on qmake (helpful for android my android port) some includes and a namespace are missing. Oddly enough, that the build works with cmake.
Unfortunately, there are also some missing includes in thirdparty libs.

lasconic commented:
I'm not really sure why we would need this in our repo since it's working. Adding more include will just increase the build time. Are you sure you use the precompile header for thirdparties too? Are you sure you really need all these third parties?

ChWick commented:
Yay, I forgot to include the PCH in the thirdparty libs.

Some information about the remaining (minimal?) changes.

Optional:
- **dsp.cpp** is not required for compiling, so I simply excluded it. Still the `namespace Ms` wrapper would be missing

Required for my qmake build:
- in **all.h** a `#ifdefined __cplusplus` is required to separate c and c++ headers (see http://doc.qt.io/qt-5/qmake-precompiledheaders.html). Else, In the 'c pass' the stl libraries are not found.

Required (and could also fail in CMake build?):
- **icon.h** undefined IconType: added mscore include due to usage of `IconType::NONE`
- **cursor.h** undefined SegmentType (line 90), ScoreElement, and access to undeclared Element: predefinition of ScoreElement, and moved functions to the source file (analog to `ElementW::get(const QString&)` ), added predefinition of enum class SegmentType
- **editstyle.h** undefined VOICES, defined in `libmscore/element.h` AND `libmscore/mscore.h` (shouldn't there only be one?) I used `libmscore/mscore.h`
- **instepctorBarline.h** undefined `InspectorElementBase` (required include), undefined `Ui::InspectorSegment` (required include)
- **selectnotedialog.h** undefined `Note` (added predefinition of `class Note`)

Do you have other suggestions?

lasconic commented:
Thanks. LGTM.
